plugins/module_utils/config_utils.py

```diff
@@ -99,3 +104,192 @@ def changed(self) -> bool:
         """
         orig_dict = self._parse_config_from_file()
         return orig_dict != self._config_dict
+
+    def _search_map(self, dictionary, key):
```

Comment:
The typing is not uniform across the code

```suggestion
    def _search_map(self, dictionary, key) -> Union[str | None]:
```

Also to make it more clear that this method is supposed to `return None` it might be nice to explicitly return None. Something like

```
def myfunc() -> Union[str | None]:
    result = None
    if foo:
        result = "bar"
    return result
```

This will also get rid of the pylint `no-else-return` / `inconsistent-return-statements` warnings

Reply:
Thanks for this suggestion. I absolutely agree with you and thank you for pointing out this inconsistency. I have ensured consistent typing across the functions.